Update make_restarted_file.sh, add G92,E fix,E safe check

sunnydrake opened the pull request:
Add G92 position reset
fix handle of E position got G0/1
check if newer E value >= 0

sunnydrake committed:
Update make_restarted_file.sh, add G92,E fix,E safe check 

Add G92 position reset
fix handle of E position got G0/1
check if newer E value >= 0

T9Air commented:
Hi, sorry for taking so long to respond, I was busy working on something else. I don't fully understand what this fixes/improves, so if you explain I would greatly appreciate it. Thanks!

sunnydrake commented:
Sorry for late reply. G92 - sets a temporary zero point (offset) for the work coordinates including E (filament/extruder position).
e.g. all other moves will be relative to this 0,0 point from code point of view.
E sometimes used by klipper as filament position or file print position so i added proper restore Eposition and zeroed and recalculated it in some cases to print from E0 position onwards not like E0 - E9912313912 like it was before.So no filament runout issue, also added proper regex and 0 checks for E number detection
